src/TConsole.cpp

```diff
@@ -837,12 +843,12 @@ void TConsole::toggleLogging(bool isMessageEnabled)
 
         mpHost->mIsCurrentLogFileInHtmlFormat = mpHost->mIsNextLogFileInHtmlFormat;
         if (mpHost->mIsCurrentLogFileInHtmlFormat) {
-            mLogFileName = QStringLiteral("%1/%2.html").arg(directoryLogFile, QDateTime::currentDateTime().toString(QStringLiteral("yyyy-MM-dd#hh-mm-ss")));
+            mLogFileName = QStringLiteral("%1/%2.html").arg(directoryLogFile, QDateTime::currentDateTime().toString(mpHost->mLogFileNameFormat));
         } else {
-            mLogFileName = QStringLiteral("%1/%2.txt").arg(directoryLogFile, QDateTime::currentDateTime().toString(QStringLiteral("yyyy-MM-dd#hh-mm-ss")));
+            mLogFileName = QStringLiteral("%1/%2.txt").arg(directoryLogFile, QDateTime::currentDateTime().toString(mpHost->mLogFileNameFormat));
         }
         mLogFile.setFileName(mLogFileName);
-        mLogFile.open(QIODevice::WriteOnly);
+        mLogFile.open(QIODevice::WriteOnly | QIODevice::Append);
```

Comment:
This won't work - you need to find the newest log file of the right type (as specified by `mpHost->mIsCurrentLogFileInHtmlFormat`) and open **that** if you are going for an **append** to a log - but not everyone will want that option - I probably won't for instance, in which case the current code is fine for that.

If you are going to do an append and the format is HTML then the existing file **must** be opened Read/Write - and you must remove everything after the last `</div>` tag in the existing file, and then modify *somehow* the current code (in the `TBuffer` class) that creates the HTML so that everything before the first `<div `...`>` tag in the new stuff is skipped in a suitable way.  This is because the last time I revised the HTML creation code I intended it to be fairly painless to combine multiple files together and put a pair of `<div>`...`</div>` tags around the entire *body* text so that each log file could be joined together with the others at those points... see https://github.com/Mudlet/Mudlet/pull/208 . You will need to do the same sort of thing.